src/settings/toml/manifest.rs

```diff
@@ -115,6 +120,35 @@ impl Manifest {
         Ok(template_config)
     }
 
+    pub fn worker_name(&self, env_arg: Option<&str>) -> String {
+        if let Some(environment) = self.get_environment(env_arg).unwrap_or_default() {
+            if let Some(name) = &environment.name {
+                return name.clone();
+            }
+            if let Some(env) = env_arg {
+                return format!("{}-{}", self.name, env);
+            }
+        }
+
+        self.name.clone()
+    }
+
+    fn route_config(&self) -> RouteConfig {
```

Comment:
route config consists entirely of the top level parts of the manifest relevant to a zoned deploy; it's sort of a constructor options method for DeployTarget as an alternative to a ridiculously long list of arguments.